### PR TITLE
Add binding for POST /repos/migrate

### PR DESCRIPTION
**Pull request by rubenv**
Relates to gogits/gogs#1843

### CODE REVIEW

File: repo.go [hunk 1]
```diff
@@ -82,3 +82,29 @@ func (c *Client) DeleteRepo(owner, repo string) error {
 	_, err := c.getResponse("DELETE", fmt.Sprintf("/repos/%s/%s", owner, repo), nil, nil)
 	return err
 }
+
+type MigrateRepoOption struct {
+	CloneAddr    string `json:"clone_addr" binding:"Required"`
```
Comment (Unknwon) [inline repo.go]: I think binding is not needed in this case :D

File: repo.go [hunk 2]
```diff
@@ -82,3 +82,29 @@ func (c *Client) DeleteRepo(owner, repo string) error {
...
+	AuthUsername string `json:"auth_username"`
+	AuthPassword string `json:"auth_password"`
+	Uid          int    `json:"uid" binding:"Required"`
```
Comment (Unknwon) [inline repo.go]: Please use `UID` as the new convention of Gogs.

File: repo.go (hunk 1 above)
Comment (rubenv) [inline repo.go]: It's used higher up in the file as well, so I'm following convention here :-)

Want me to remove it?

Comment (Unknwon) [inline repo.go]: Sorry, my mistake...

**Comment by Unknwon** [comment]
Thanks!

**Pull request merged**